shell/packages/sandstorm-permissions/permissions.js

```diff
@@ -256,18 +256,17 @@ class RequirementSet {
   }
 }
 
-const permissionIdPattern = Match.OneOf({ canAccess: null }, { appDefined: Match.Integer });
-// As we compute a flow of permissions, we need to be able to explicitly refer to not only
-// the permissions enumerated in a grain's `ViewInfo`, but also the usually-implicit "can
-// access the grain at all" permission. A permission ID, defined here, allows us to refer
-// to either of these types of permissions.
+// A permission ID is a stringified integer greater than or equal to -1. The permission "-1"
```

Comment:
Stringified because as far as I understand, keys in plain javascript objects are always strings. I considered using an `Array` rather than a plain object here, but that seemed weird.

-1 because that makes it somewhat easier to iterate through the permissions and to find the maximum key.